api/endpoint-tests/roles/delete.js

```diff
@@ -0,0 +1,72 @@
+const tap = require('tap'); // eslint-disable-line import/no-extraneous-dependencies
+const { db, request, getFullPath, login } = require('../utils');
+
+tap.test('roles endpoint | DELETE /roles/:roleID', async deleteRolesTests => {
+  await db().seed.run();
+
+  const url = getFullPath('/roles');
+
+  deleteRolesTests.test('when unauthenticated', async unauthenticatedTests => {
+    unauthenticatedTests.test('with invalid ID', async invalidTest => {
+      const { response, body } = await request.delete(`${url}/9001`);
+      invalidTest.equal(response.statusCode, 403, 'gives a 403 status code');
+      invalidTest.notOk(body, 'does not send a body');
+    });
+
+    unauthenticatedTests.test('with invalid ID', async invalidTest => {
+      const { response, body } = await request.delete(`${url}/1`);
+      invalidTest.equal(response.statusCode, 403, 'gives a 403 status code');
+      invalidTest.notOk(body, 'does not send a body');
+    });
+  });
+
+  deleteRolesTests.test('when authenticated', async authenticatedTests => {
+    const cookies = await login();
+
+    authenticatedTests.test('with an invalid ID', async invalidTest => {
+      const { response, body } = await request.delete(`${url}/9001`, {
+        jar: cookies
+      });
+      invalidTest.equal(response.statusCode, 404, 'gives a 404 status code');
+      invalidTest.notOk(body, 'does not send a body');
+    });
+
+    authenticatedTests.test(
+      'with an ID for a role that the user belongs to',
```

Comment:
can u explain this test to me and why we want 401 here?

Reply:
This is testing the case where a user attempts to delete the role that they're in.  They shouldn't be allowed to do that because they'd end up locking themselves out of the system, so it's a 401 (unauthorized).

This is related to #222 to go back and modify the PUT route to prevent users from modifying the roles they belong to, too.  I figured since I was writing this code at the time, I'd go ahead and fix it here.  But your question is a good one, and I think I should update the test name to be more descriptive.

Reply:
ok, i see and that makes perfect sense -- thanks for explanation!